tests/scenarios/core-resolver-test.ts

```diff
@@ -845,12 +858,15 @@ Scenarios.fromProject(() => new Project())
 
           let switcherModule = expectAudit.module('./app.js').resolves('my-app/hello-world').toModule();
           switcherModule.codeEquals(`
-            import { macroCondition, getGlobalConfig, importSync } from '@embroider/macros';
+            import { macroCondition as macroCondition0, getGlobalConfig as getGlobalConfig0 } from "../node_modules/@embroider/compat/node_modules/@embroider/macros/src/addon/runtime";
+            import esc from "../node_modules/@embroider/compat/node_modules/@embroider/macros/src/addon/es-compat2";
+            import * as _importSync20 from "./fastboot";
+            import * as _importSync40 from "./browser";
```

Comment:
Promotion of importSync to the top-level outside of the macroCondition is apparently the current and expected beahvior, even though it can lead to problematic double imports like this.

The [solution](https://discord.com/channels/480462759797063690/568935504288940056/1383089548870815804) per @ef4 is to use `await import(...)` in place of [importSync](https://github.com/embroider-build/embroider/blob/91b124a/packages/core/src/virtual-content.ts#L132).  That is an unrelated problem that already exists, and it makes sense to address it (and [other `importSync` usages](https://github.com/search?q=repo%3Aembroider-build%2Fembroider+importSync+path%3A%2F%5Epackages%5C%2Fcompat%5C%2Fsrc%5C%2F%2F&type=code)) separately.

Reply:
Per today's meeting, this test should not actually be transpiling the macros, so I'm backing this out.